core/src/main/java/com/tapptitude/core/repository/ImageRepositoryImpl.kt

```diff
@@ -1,12 +1,23 @@
 package com.tapptitude.core.repository
 
 import com.tapptitude.core.model.Image
+import com.tapptitude.core.model.extension.toEntity
 import com.tapptitude.core.model.extension.toImage
 import com.tapptitude.network.source.ImageRemoteDataSource
+import com.tapptitude.persistence.images.ImagesDao
+import kotlinx.coroutines.Dispatchers.IO
+import kotlinx.coroutines.withContext
 
-class ImageRepositoryImpl(private val imageRemoteDataSource: ImageRemoteDataSource) : ImageRepository {
-    override suspend fun getRandomImage(): Image {
-        return imageRemoteDataSource.getRandomImage()
-            .toImage()
+class ImageRepositoryImpl(
+    private val imageRemoteDataSource: ImageRemoteDataSource,
+    private val imagesDao: ImagesDao
+) : ImageRepository {
+
+    override suspend fun getRandomImage(): Image = withContext(IO) {
```

Comment:
I think we should hide the context behind abstractions, or even inject them.
I disagree with using components from the Kotlin API directly, because the connections are too tight.

I would suggest having another class, something like `DemoDispatchers`, that has all the dispatchers we need:
- one for IO
- one for Main
- one for Main.immediate
- and maybe one more ?

We could inject this class everywhere we need dispatchers and use them, thus not depending on the platform directly

Reply:
I'm not sure what benefit this would provide if any and whether it is or not worth it

Reply:
to be discussed as this might need a separate module (utils or something)